### PR TITLE
Reset no longer needed on staking & fix underflow

### LINKED ISSUE #108: CS-V2Gov-041: Remaining ToDos in the Codebase
There are remaining TODO and `@audit` comments in the codebase. Addressing remaining notes help improve the quality and readability of the code.

---

#### Code partially corrected:
The previous comments have been removed from the contracts in scope, however, a new TODO comment is present in function `_increaseUserVoteTrackers()`.

### LINKED ISSUE #109: CS-V2Gov-050: Possible Underflow in `withdrawLQTY`
Function `withdrawLQTY()` updates user's state with the new offset and unallocated LQTY as follows:

```solidity
// Update the offset tracker
if (_lqtyAmount < userState.unallocatedLQTY) {
    // ...
} else {
    // ...
    userState.unallocatedOffset = 0;
}

// Update the user's LQTY tracker
userState.unallocatedLQTY -= _lqtyAmount;
```

The function doesn't perform any check that amount of LQTY being withdrawn is less than the user's stake. Furthermore, `LQTYStaking.unstake()` caps the amount to the user's stake. Therefore, if one calls `withdrawLQTY()` with a larger LQTY amount than staked, the subtraction above reverts with an underflow error.

### PR DESCRIPTION
**Pull request by danielattilasimon**
Address remaining TODO in `_increaseUserVoteTrackers()` by eliminating reset requirement.
Fixes #108.

Fix possible underflow in `withdrawLQTY()`.
Fixes #109.

### CODE REVIEW

**Review by bingen — APPROVED** [review]
Looks good!
I think there are some other places where we may update tests to the fact that we don’t need resets on deposit/withdrawal anymore, like:

https://github.com/liquity/V2-gov/blob/main/test/recon/targets/GovernanceTargets.sol#L110
https://github.com/liquity/V2-gov/blob/main/test/recon/targets/GovernanceTargets.sol#L283

But happy to merge this as is, we can fix that later if it’s more convenient.

File: src/Governance.sol [hunk 1]
```diff
@@ -203,7 +196,6 @@ contract Governance is MultiDelegateCall, UserProxyFactory, ReentrancyGuard, Own
     function withdrawLQTY(uint256 _lqtyAmount, bool _doSendRewards, address _recipient) public nonReentrant {
         // check that user has reset before changing lqty balance
```
[the remark this answers is not in the document]
Comment (danielattilasimon) [reply src/Governance.sol:197]: Oops 😅

**Comment by danielattilasimon** [comment]
I'd fix the invariant tests in a separate PR later, as there are already some broken ones after the voting power refactor.

**Comment by danielattilasimon** [comment]
On the other hand, I dug up the test case I wrote that was demonstrating the bug that was the whole reason for the reset requirement, updated it, and added it to the test suite.

**Comment by bingen** [comment]
Awesome, thanks!!

**Pull request merged**